### PR TITLE
Databricks Custom Policies

### PR DESCRIPTION
**Pull request by marvinbuss**
## Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

1. Added Custom Databricks Policies

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Comment by marvinbuss** [comment]
@krnese Can you please review this structure?

File: eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-NoPublicIp.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                        ]
+                    },
+                    "then": {
+                        "effect": "[[parameters('effect')]"
+                    }
+                },
+                "metadata": {
+                    "version": "1.0.0",
+                    "category": "Databricks",
+                    "preview": false,
```
Comment (krnese) [inline eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-NoPublicIp.json:51]: Can we just drop these from the metadata object?

File: eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "resources": [
+        {
+            "type": "Microsoft.Authorization/policyDefinitions",
+            "apiVersion": "2020-03-01",
+            "name": "Deny-Databricks-Sku",
+            "properties": {
+                "policyType": "Custom",
+                "mode": "Indexed",
+                "displayName": "Deny non-premium databricks sku",
+                "description": "Enfoces the use of Premium Databricks workspaces.",
```
Comment (krnese) [inline eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json:15]: Pls expand on the description per our discussion, to highlight this is to ensure appropriate security features, RBAC etc.

File: eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    },
+                    "then": {
+                        "effect": "[[parameters('effect')]"
+                    }
+                },
+                "metadata": {
+                    "version": "1.0.0",
+                    "category": "Databricks",
+                    "preview": false,
+                    "deprecated": false
```
Comment (krnese) [inline eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json:52]: lets drop "preview" and "deprecated" in the metadata object

**Review by krnese — CHANGES_REQUESTED** [review]

File: eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json (hunk 3 above)
Comment (marvinbuss) [reply eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json]: DONE

File: eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json (hunk 2 above)
Comment (marvinbuss) [reply eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-Sku.json]: DONE

File: eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-NoPublicIp.json (hunk 1 above)
Comment (marvinbuss) [reply eslzArm/managementGroupTemplates/policyDefinitions/Databricks/DENY-Databricks-NoPublicIp.json]: DONE

**Review by krnese — APPROVED** [review]

**Pull request merged**